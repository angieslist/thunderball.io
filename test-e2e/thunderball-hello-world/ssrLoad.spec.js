import fetch from 'node-fetch';
import Redis from 'ioredis';

const redis = new Redis({ host: 'localhost' });

function normalizeText(text) {
  // normalize hashed assets
  return text.replace(/[\d\w]{7,}(?=\.js|\.css)/g, 'TEST-HASH')
    // normalize app version
    .replace(/appVersion":"\d+\.\d+\.\d+/, 'appVersion":"TEST-VERSION')
    // normalize data-react-checksum
    .replace(/-?\d{7,}/g, 'TEST-CHECKSUM');
}

describe('ssrLoad', () => {
  describe('Get page that uses ssrLoad', () => {
    it('returns the page', async () => {
      const response = await fetch('http://localhost:8000/todo-answers/server-fetch-example');
      expect(response).toHaveProperty('status', 200);
      const responseText = await response.text();
      const normalizedText = normalizeText(responseText);
      const cachedHtmlAndStatus = await redis.get('/todo-answers/server-fetch-example');
      expect(cachedHtmlAndStatus).toBe(`200${responseText}`);
      expect(normalizedText).toMatchSnapshot();
    });
  });
});
