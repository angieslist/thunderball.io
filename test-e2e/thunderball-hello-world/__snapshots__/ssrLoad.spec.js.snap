// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssrLoad Get page that uses ssrLoad returns the page 1`] = `

<!DOCTYPE html>
<html lang="en">
  <head>
    <title data-rh="true">
      Server Fetch Todos - Thunderball Hello World
    </title>
    <link data-rh="true"
          rel="shortcut icon"
          type="image/png"
          href="/assets/fb8bTEST-CHECKSUMe246c3363bfcc4f601.png"
    >
    <link data-rh="true"
          rel="stylesheet"
          href="//fonts.googleapis.com/css?family=Roboto:300,300italic,700,700italic"
    >
    <link data-rh="true"
          rel="stylesheet"
          href="//cdn.rawgit.com/necolas/TEST-HASH.css/master/TEST-HASH.css"
    >
    <link data-rh="true"
          rel="stylesheet"
          href="//cdn.rawgit.com/milligram/milligram/master/dist/milligram.min.css"
    >
    <link href="/assets/commons-TEST-HASH.css"
          rel="stylesheet"
          key="0"
    >
  </head>
  <body>
    <script type="text/javascript">
      window.__INITIAL_STATE__ = {"config":{"appVersion":"TEST-VERSION","appName":"thunderball-hello-world"},"device":{"host":"http:\\u002F\\u002Flocalhost:8000"},"routing":{"locationBeforeTransitions":{"pathname":"\\u002Ftodo-answers\\u002Fserver-fetch-example","search":"","hash":"","action":"POP","key":null,"query":{}}},"todos":[{"id":1,"text":"Learn React","completed":true},{"id":2,"text":"Learn Thunderball","completed":false}],"visibilityFilter":"SHOW_ALL"};
    </script>
    <div id="app">
      <div data-reactroot>
        <div class="container">
          <div class="row">
            <div class="column">
              <div>
                <div>
                  <h1>
                    Todos
                  </h1>
                  <div>
                    <div>
                      <input type="checkbox"
                             id="todo-1"
                             checked
                      >
                      <label class="label-inline"
                             for="todo-1"
                      >
                        Learn React
                      </label>
                    </div>
                    <div>
                      <input type="checkbox"
                             id="todo-2"
                      >
                      <label class="label-inline"
                             for="todo-2"
                      >
                        Learn Thunderball
                      </label>
                    </div>
                  </div>
                  <hr>
                  <a href="/todo-answers">
                    Todos
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <script type="text/javascript"
            src="https://cdn.polyfill.io/v2/polyfill.min.js?features=Intl.~locale.en"
    >
    </script>
    <script type="text/javascript"
            src="/assets/commons-TEST-HASH.js"
    >
    </script>
    <script type="text/javascript"
            src="/assets/todos-answers-TEST-HASH.js"
    >
    </script>
  </body>
</html>

`;
